packages/nextjs/src/server/protect.ts

```diff
@@ -112,10 +112,8 @@ export const createProtect = (opts: {
 
 const isServerActionRequest = (req: Request) => {
   return (
-    !!req.headers.get(nextConstants.Headers.NextUrl) &&
-    (req.headers.get(constants.Headers.Accept)?.includes('text/x-component') ||
-      req.headers.get(constants.Headers.ContentType)?.includes('multipart/form-data') ||
```

Comment:
Checking against multipart is not realistic. The following example will fire a request with Content-type `text/plain;charset=UTF-8`

```tsx
'use client';

import { addItem2 } from '../actions';

export default function AddToCart() {
  return (
    <button
      type='button'
      onClick={() => {
        addItem2({ wowowwo: 'dwa' });
      }}
    >
      Somethj
    </button>
  );
}

```